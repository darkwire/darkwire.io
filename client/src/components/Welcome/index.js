import React, { Component } from 'react'
import PropTypes from 'prop-types'
import RoomLink from 'components/RoomLink'

class Welcome extends Component {
  constructor(props) {
    super(props)
    this.state = {
      roomUrl: `https://darkwire.io/${props.roomId}`,
    }
  }

  render() {
    return (
      <div>
        <div>
          v2.0 is a complete rewrite and includes several new features. Here are some highlights:
          <ul className="native">
            <li>Support on all modern browsers (Chrome, Firefox, Safari, Safari iOS, Android)</li>
            <li>Slash commands (/nick, /me, /clear)</li>
            <li>Room owners can lock the room, preventing anyone else from joining</li>
            <li>Front-end rewritten in React.js and Redux</li>
            <li>Send files up to 4 MB</li>
          </ul>
          <div>
            You can learn more <a href="https://github.com/darkwire/darkwire.io" target="_blank" rel="noopener noreferrer">here</a>.
          </div>
        </div>
        <br />
        <p className='mb-2'>Others can join this room using the following URL:</p>
        <RoomLink roomId={this.props.roomId} translations={this.props.translations} />
        <div className="react-modal-footer">
          <button className="btn btn-primary btn-lg" onClick={this.props.close}>{this.props.translations.welcomeModalCTA}</button>
        </div>
      </div>
    )
  }
}

Welcome.propTypes = {
  roomId: PropTypes.string.isRequired,
  close: PropTypes.func.isRequired,
  translations: PropTypes.object.isRequired,
}

export default Welcome
