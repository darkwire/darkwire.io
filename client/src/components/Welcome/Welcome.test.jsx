import React from 'react';
import { render } from '@testing-library/react';
import { test, expect, vi } from 'vitest';

import Welcome from '.';

vi.mock('@/components/RoomLink');

test('Welcome component is displaying', async () => {
  const { asFragment } = render(<Welcome roomId="roomtest" close={() => {}} translations={{}} />);

  expect(asFragment()).toMatchSnapshot();
});
