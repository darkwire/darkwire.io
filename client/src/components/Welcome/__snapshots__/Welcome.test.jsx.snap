// Vitest Snapshot v1

exports[`Welcome component is displaying 1`] = `
<DocumentFragment>
  <div>
    <div>
      v2.0 is a complete rewrite and includes several new features. Here are some highlights:
      <ul
        class="native"
      >
        <li>
          Support on all modern browsers (Chrome, Firefox, Safari, Safari iOS, Android)
        </li>
        <li>
          Slash commands (/nick, /me, /clear)
        </li>
        <li>
          Room owners can lock the room, preventing anyone else from joining
        </li>
        <li>
          Front-end rewritten in React.js and Redux
        </li>
        <li>
          Send files up to 4 MB
        </li>
      </ul>
      <div>
        You can learn more 
        <a
          href="https://github.com/darkwire/darkwire.io"
          rel="noopener noreferrer"
          target="_blank"
        >
          here
        </a>
        .
      </div>
    </div>
    <br />
    <p
      class="mb-2"
    >
      Others can join this room using the following URL:
    </p>
    <div
      class="react-modal-footer"
    >
      <button
        class="btn btn-primary btn-lg"
      />
    </div>
  </div>
</DocumentFragment>
`;
